### PR TITLE
[Tier0][Test] Validate package and image version

### PR DESCRIPTION
**Pull request by adityaramteke**
This PR is to validate the package and image version for release. 

The following steps are performed within the test:

Step 1: Check installed packages
Step 2: Check installed package versions
Step 3: Check for dependent packages
Step 4: Install cephadm-ansilbe package and check version
Step 5: Check dependent package for cephadm-ansible
Step 6: Check Ceph client packages and version (Check the document)
Step 7: Check dependent package for cephadm-ansible
Step 8: Pull the latest monitoring stack images (ceph-dashboard, alert-manager, grafana, node-exporter, prometheus)
Step 9: Inspect monitoring stack images
Step 11: Check the image version
Step 12: Cleanup cephadm-ansible and client packages

Full Logs:
Quincy: http://magna002.ceph.redhat.com/cephci-jenkins/aramteke/package-version/quincy/cephci-run-38B5WQ/

### CODE REVIEW

File: cli/utilities/containers.py [hunk 1]
```diff
@@ -136,3 +139,35 @@ def pull(self, image):
...
         if self.execute(sudo=True, long_running=True, cmd=cmd):
             raise ContainerRegistryError(f"Failed to pull container image '{image}'")
+
+    def rmi(self, image):
+        """Remove container image
+
+        Args:
+            image (str): Container image url
+        """
+
```
Comment (vamahaja) [inline cli/utilities/containers.py:149]: Remove this extra line

File: cli/utilities/containers.py [hunk 2]
```diff
@@ -136,3 +139,35 @@ def pull(self, image):
...
+        cmd = f"{self.base_cmd} rmi {image}"
+
```
Comment (vamahaja) [inline cli/utilities/containers.py:151]: remove this extra line

File: tests/cephadm/test_package_validation.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+]
+
+rpm_client_packages = [
+    "ceph-common",
+    "ceph-fuse",
+    "cephfs-top",
+]
+
+
+def validate_rpm_version(node, package_list, version_config):
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:40]: better to move this to common space

File: tests/cephadm/test_package_validation.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        version = re.search(rpm_regex, out).group(1)
+        if version < version_config.get(pkg):
+            raise PackageError(
+                f"Expected version for {pkg} is {version_config.get(pkg)}; found {version}"
+            )
+        else:
+            log.info(f"{pkg} installed correctly with version {version}")
+
+
+def validate_image_version(node, image_dict, version_config):
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:63]: same as above

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: cli/utilities/packages.py [hunk 5]
```diff
@@ -83,6 +83,30 @@ def install(self, pkg, nogpgcheck=False, env_vars={}):
...
+
+        Args:
+            pkg (str): package needs to be removed
+            env_vars (dict): dictiory with environment variables
+        """
+
+        cmd = ""
+        if env_vars:
+            for k, v in env_vars.items():
+                cmd += f"{k}={v} "
```
[the remark this answers is not in the document]
Comment (adityaramteke) [reply cli/utilities/packages.py]: Don't fully satisfy the need. 
Here I'm trying to pass key and value as "key"="value" to cmd. 
Not trying to separate "="

File: cli/utilities/containers.py (hunk 2 above)
Comment (adityaramteke) [reply cli/utilities/containers.py]: Done

File: tests/cephadm/test_package_validation.py (hunk 4 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:63]: @vamahaja do you mean I should it to lib?

File: tests/cephadm/test_package_validation.py [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+        else:
+            log.info(f"{image} installed correctly with version {version}")
+
+
+def run(ceph_cluster, **kwargs):
+    """Verify package version validation
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:85]: Remove extra line

File: tests/cephadm/test_package_validation.py [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+def validate_rpm_version(node, package_list, version_config):
+    """Validate installed rpm package version
+
+    Args:
+        node (ceph.ceph.CephNode): Node name
+        package_list (list): List of packages
+        version_config (dict): List of packages and package version
+    """
+    # Define regex to get version
+    rpm_regex = r"Version\s+:\s+(\d+\.\d+)"
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:45]: use `_regex`

File: tests/cephadm/test_package_validation.py [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+]
+
+rpm_client_packages = [
+    "ceph-common",
+    "ceph-fuse",
+    "cephfs-top",
+]
+
+
+def validate_rpm_version(node, package_list, version_config):
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:36]: Create this method for single package instead of list

Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:36]: Add method as compare here [1], which will return -1, 0, 1 based on comparision.

[1] https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/packages.py#L23

File: tests/cephadm/test_package_validation.py (hunk 7 above)
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:45]: Is this common version regex ?

File: tests/cephadm/test_package_validation.py (hunk 4 above)
Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:59]: Same as above add here - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/containers.py#L47

File: tests/cephadm/test_package_validation.py [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+def run(ceph_cluster, **kwargs):
+    """Verify package version validation
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get installer and client node
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+    client = ceph_cluster.get_nodes(role="client")[0]
+    log.info(f"Check type: {type(installer)}")
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:89]: Add apropriate info log

File: tests/cephadm/test_package_validation.py [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+
+    # Check if requied packages are installed
+    for pkg in installed_packages:
+        if not Rpm(installer).query(pkg):
+            raise PackageError(f"Required package {pkg} is not installed")
+
+    # Get config details
+    config = kwargs.get("config")
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:97]: Get these details at the start of the method

File: tests/cephadm/test_package_validation.py [hunk 11]
```diff
@@ -0,0 +1,162 @@
...
+    cloud_type = config.get("cloud_type")
+
+    # Enable ceph tools repo on installer and client
+    base_url = get_custom_repo_url(base_url, cloud_type)
+    for node in {installer, client}:
+        if not Package(node).add_repo(base_url):
+            raise ReposError(f"Failed to enable {base_url} repo")
+
+    # Get the Tools repo file name
+    out = installer.get_dir_list(dir_path=YUM_REPO_DIR)
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:108]: Why this additional check is required ? If add repo is sucessful then there should be repo file added to this path

File: tests/cephadm/test_package_validation.py [hunk 12]
```diff
@@ -0,0 +1,162 @@
...
+    file_name = [repo for repo in out if "Tools" in repo][0]
+
+    # Get expected package versions from config
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:111]: Move this to start of the method

File: tests/cephadm/test_package_validation.py [hunk 13]
```diff
@@ -0,0 +1,162 @@
...
+    rpm_version = config.get("rpm_package_version")
+    rpm_client_version = config.get("rpm_client_package_version")
+    rpm_dependent_package_version = config.get("rpm_dependent_package_version")
+    container_version = config.get("container_image_version")
+    packages = installed_packages + packages_to_install
+
+    # Get monitoring stack images
+    monitoring_images = config.get("monitoring_images")
+
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:120]: Remove extra line

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/cephadm/test_package_validation.py (hunk 9 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:89]: I'll remove it, it doesn't require

File: tests/cephadm/test_package_validation.py (hunk 10 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:97]: I think this placement is correct at this step. 

As we are checking whether installed packages are present or not before getting the config information. If installed required packages are not present in the node these configs are useless. 

Please check again and let me know.

File: tests/cephadm/test_package_validation.py (hunk 11 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:108]: Yes, this is to get the name of repo file which will be used to remove at the finally block.

File: tests/cephadm/test_package_validation.py (hunk 12 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:111]: Again this placement is correct after we enable tools repo. If tools repos enable will fail there is no use of getting the config file.

File: tests/cephadm/test_package_validation.py (hunk 13 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:120]: There is no extra line here, we can keep it.

File: tests/cephadm/test_package_validation.py (hunk 6 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:85]: Sure will do

File: tests/cephadm/test_package_validation.py (hunk 8 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:36]: I didn't understand why we needed to do it for a single package.

As per the test requirement, this method makes it easy to use with package_list as we are assigning most of the packages in the list.

File: tests/cephadm/test_package_validation.py (hunk 7 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:45]: Yes, this is common version regex.

Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:45]: I think we can keep the name as rpm_regex itself. As if required for container we can distinct as conatiner_regex.

File: tests/cephadm/test_package_validation.py (hunk 10 above)
Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:97]: All these are mandatory parameters, better to have validation at the start

File: tests/cephadm/test_package_validation.py (hunk 11 above)
Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:108]: got it

File: tests/cephadm/test_package_validation.py (hunk 7 above)
Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:45]: This won't be added in the same method and method name itself have rpm

File: tests/cephadm/test_package_validation.py (hunk 8 above)
Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:36]: this is comparision and for each package same steps need to be repeated. Instead loop inside method, loop over the method.

As this is specific to comparing rpm package version, we already have class Package which perform respective operations.

Add same steps to compare version and return below codes as described -

1. `-1` if version is less than expected
2. `0` if version is exactly same
3. `1` if version is more than expected

File: tests/cephadm/test_package_validation.py (hunk 10 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:90]: Done, updated.

File: suites/pacific/cephadm/tier-0.yaml [hunk 14]
```diff
@@ -17,6 +17,34 @@ tests:
...
+      desc: Validate package versions
+      module: test_package_validation.py
+      name: Validate package versions
+      config:
+        monitoring_images:
+         node_exporter: registry.redhat.io/openshift4/ose-prometheus-node-exporter:latest
+         prometheus: registry.redhat.io/openshift4/ose-prometheus:latest
+         alertmanager: registry.redhat.io/openshift4/ose-prometheus-alertmanager:latest
+        rpm_package_version:
+          cephadm-ansible: "1.17"
```
Comment (vamahaja) [inline suites/pacific/cephadm/tier-0.yaml:31]: this should come under `ceph_packages`

File: suites/quincy/cephadm/tier-0.yaml [hunk 15]
```diff
@@ -17,6 +17,34 @@ tests:
       module: install_prereq.py
       name: setup pre-requisites
 
+  - test:
+      abort-on-fail: true
+      desc: Validate package versions
+      module: test_package_validation.py
+      name: Validate package versions
+      config:
+        monitoring_images:
```
Comment (vamahaja) [inline suites/quincy/cephadm/tier-0.yaml:26]: Same as bove config

File: suites/reef/cephadm/tier-0.yaml [hunk 16]
```diff
@@ -17,6 +17,34 @@ tests:
       module: install_prereq.py
       name: setup pre-requisites
 
+  - test:
+      abort-on-fail: true
+      desc: Validate package versions
+      module: test_package_validation.py
+      name: Validate package versions
+      config:
+        monitoring_images:
```
Comment (vamahaja) [inline suites/reef/cephadm/tier-0.yaml:26]: Same as above configs

File: tests/cephadm/test_package_validation.py [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+from cli.utilities.packages import Package, PackageError, ReposError, Rpm
+from cli.utilities.utils import get_custom_repo_url
+from utility.log import Log
+
+log = Log(__name__)
+
+
+YUM_REPO_DIR = "/etc/yum.repos.d"
+
+installed_pkgs = [
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:11]: No need to maintain list here. Use keys passed in configs as package names

File: tests/cephadm/test_package_validation.py [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+]
+
+rpm_client_pkgs = [
+    "ceph-common",
+    "ceph-fuse",
+    "cephfs-top",
+]
+
+
+def run(ceph_cluster, **kwargs):
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:34]: I would suggest to create separate methods to validate these packages with finally method having cleanup

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: suites/pacific/cephadm/tier-0.yaml (hunk 14 above)
Comment (adityaramteke) [reply suites/pacific/cephadm/tier-0.yaml:27]: Updated the comment.

File: suites/pacific/cephadm/tier-0.yaml [hunk 19]
```diff
@@ -17,6 +17,34 @@ tests:
...
+          chrony: "4.1"
```
[the remark this answers is not in the document]
Comment (adityaramteke) [reply suites/pacific/cephadm/tier-0.yaml:29]: Updated name

File: tests/cephadm/test_package_validation.py (hunk 18 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py:100]: Updated the changes.

File: tests/cephadm/test_package_validation.py [hunk 20]
```diff
@@ -0,0 +1,156 @@
...
+from cli.utilities.utils import get_custom_repo_url
+from utility.log import Log
+
+log = Log(__name__)
+
+
+YUM_REPO_DIR = "/etc/yum.repos.d"
+
+
+def check_server_packages(node, server_pkgs):
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:12]: `validate_*`

Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:12]: `server_pkgs` == `pkgs`

File: tests/cephadm/test_package_validation.py [hunk 21]
```diff
@@ -0,0 +1,156 @@
...
+        server_pkgs (dict): server packages and version
+    """
+    try:
+        # Install required ceph packages on installer
+        for pkg, _ in server_pkgs.items():
+            if not Rpm(node).query(pkg):
+                Package(node).install(pkg, nogpgcheck=True)
+
+        # Check expected package version
+        for pkg, ver in server_pkgs.items():
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:25]: Can be done in one loop. Install package and then check for version.

File: tests/cephadm/test_package_validation.py [hunk 22]
```diff
@@ -0,0 +1,156 @@
...
+            if Package(node).compare(pkg, ver) == -1:
+                raise PackageError(f"Expected version for {pkg} is {ver}")
+    finally:
+        # Remove installed packages
+        for pkg, _ in server_pkgs.items():
+            Package(node).remove(pkg)
+
+
+def check_client_packages(node, client_pkgs):
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:34]: `client_pkgs` = `pkgs`

File: tests/cephadm/test_package_validation.py [hunk 23]
```diff
@@ -0,0 +1,156 @@
...
+        node (ceph): ceph node objects
+        client_pkgs (dict): client packages and version
+    """
+    try:
+        # Install client packages on client node
+        for pkg, _ in client_pkgs.items():
+            if not Rpm(node).query(pkg):
+                Package(node).install(pkg, nogpgcheck=True)
+
+        # Check expected package version on client
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:46]: Same as above

File: tests/cephadm/test_package_validation.py [hunk 24]
```diff
@@ -0,0 +1,156 @@
...
+        for pkg, ver in client_pkgs.items():
+            if Package(node).compare(pkg, ver) == -1:
+                raise PackageError(f"Expected version for {pkg} is {ver}")
+    finally:
+        # Remove installed packages from client
+        for pkg, _ in client_pkgs.items():
+            Package(node).remove(pkg)
+
+
+def check_default_packages(node, required_pkgs):
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:56]: `client_pkgs` == `pkgs`

File: tests/cephadm/test_package_validation.py [hunk 25]
```diff
@@ -0,0 +1,156 @@
...
+        node (ceph): ceph node objects
+        required_pkgs (dict): required packages and version
+    """
+    # Check expected package version
+    for pkg, ver in required_pkgs.items():
+        if Package(node).compare(pkg, ver) == -1:
+            raise PackageError(f"Expected version for {pkg} is {ver}")
+
+
+def check_dependent_packages(node, dependent_pkgs):
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:68]: `dependent_pkgs` == `pkgs`

File: tests/cephadm/test_package_validation.py (hunk 21 above)
Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:25]: validate_*

File: tests/cephadm/test_package_validation.py (hunk 22 above)
Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:34]: validate_*

File: tests/cephadm/test_package_validation.py (hunk 24 above)
Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:56]: validate_*

File: tests/cephadm/test_package_validation.py (hunk 25 above)
Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:68]: validate_*

File: tests/cephadm/test_package_validation.py [hunk 26]
```diff
@@ -0,0 +1,156 @@
...
+        node (ceph): ceph node objects
+        dependent_pkgs (dict): dependent packages and version
+    """
+    # Check expected package version
+    for pkg, ver in dependent_pkgs.items():
+        if Package(node).compare(pkg, ver) == -1:
+            raise PackageError(f"Expected version for {pkg} is {ver}")
+
+
+def check_monitoring_images(node, monitoring_images):
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:80]: validate_*

Comment (vamahaja) [reply tests/cephadm/test_package_validation.py:80]: `monitoring_images` == `images`

File: tests/cephadm/test_package_validation.py [hunk 27]
```diff
@@ -0,0 +1,156 @@
...
+    cloud_type = config.get("cloud_type")
+
+    # Get expected package versions from config
+    client_pkgs = config.get("client_packages")
+    dependent_pkgs = config.get("dependent_packages")
+    monitoring_images = config.get("monitoring_images")
+    required_pkgs = config.get("default_packages")
+    server_pkgs = config.get("server_packages")
+
+    # Check if requied packages are installed
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:121]: Create methos for this one and add in below `try..catch` block

File: tests/cephadm/test_package_validation.py [hunk 28]
```diff
@@ -0,0 +1,156 @@
...
+
+        # Install and check client packages
+        check_client_packages(client, client_pkgs)
+
+        # Check expected image version
+        check_monitoring_images(installer, monitoring_images)
+
+    finally:
+        # Remove repo file from node
+        installer.remove_file(file_path=f"{YUM_REPO_DIR}/{file_name}")
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:154]: Add step for `yum clean` to delete cache

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/cephadm/test_package_validation.py (hunk 22 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py]: Updated changes

File: tests/cephadm/test_package_validation.py (hunk 23 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py]: Updated changes

File: tests/cephadm/test_package_validation.py (hunk 24 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py]: Updated changes

File: tests/cephadm/test_package_validation.py (hunk 25 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py]: Updated changes

File: tests/cephadm/test_package_validation.py (hunk 26 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py]: Updated changes

File: tests/cephadm/test_package_validation.py (hunk 27 above)
Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py]: Updated changes

File: tests/cephadm/test_package_validation.py [hunk 29]
```diff
@@ -0,0 +1,172 @@
...
+        remove_packages(node, pkgs)
+
+
+def validate_images(node, images):
+    """Validate the monitoring images
+    Args:
+        node (ceph): ceph node objects
+        pkgs (dict): monitoring images and version
+    """
+    reg_image = "registry.redhat.io/openshift4/{}:latest"
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:108]: create global variable

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: suites/pacific/smoke/build.yaml [hunk 30]
```diff
@@ -0,0 +1,171 @@
+#===============================================================================================
+#-------------------------------------
+#-----    Smoke Test suite     ------
+#-------------------------------------
+# Conf: conf/pacific/cephadm/tier-0.yaml
```
Comment (vamahaja) [inline suites/pacific/smoke/build.yaml:5]: tier-0 ?

File: suites/pacific/smoke/build.yaml [hunk 31]
```diff
@@ -0,0 +1,171 @@
...
+        client_packages:
+          ceph-common: "16.2"
+          ceph-fuse: "16.2"
+          cephfs-top: "16.2"
+        monitoring_images:
+          ose-prometheus-node-exporter: "v4.13.0"
+          ose-prometheus: "v4.13.0"
+          ose-prometheus-alertmanager: "v4.13.0"
+
+  - test:
```
Comment (vamahaja) [inline suites/pacific/smoke/build.yaml:45]: These tests are not required.

File: suites/quincy/smoke/build.yaml [hunk 32]
```diff
@@ -0,0 +1,170 @@
...
+#-------------------------------------
+# Conf: conf/quincy/cephadm/tier-0.yaml
+# Smoke test cases for
+#    - Bootstrap
+#    - Host management
+#    - Ceph role Service deployment,
+#    - Configure client for RGW and RBD systems
+#
+#===============================================================================================
+tests:
```
Comment (vamahaja) [inline suites/quincy/smoke/build.yaml:13]: same as above

File: suites/reef/smoke/build.yaml [hunk 33]
```diff
@@ -0,0 +1,170 @@
...
+#-------------------------------------
+# Conf: conf/reef/cephadm/tier-0.yaml
+# Smoke test cases for
+#    - Bootstrap
+#    - Host management
+#    - Ceph role Service deployment,
+#    - Configure client for RGW and RBD systems
+#
+#===============================================================================================
+tests:
```
Comment (vamahaja) [inline suites/reef/smoke/build.yaml:13]: same as above

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: suites/pacific/smoke/build.yaml (hunk 30 above)
Comment (vamahaja) [reply suites/pacific/smoke/build.yaml:5]: My bad, it's config

File: suites/pacific/smoke/build.yaml (hunk 31 above)
Comment (vamahaja) [reply suites/pacific/smoke/build.yaml:45]: My bad, it's config

File: tests/cephadm/test_package_validation.py [hunk 34]
```diff
@@ -0,0 +1,170 @@
+from cli.utilities.containers import Container
+from cli.utilities.packages import Package, PackageError, ReposError, Rpm
+from cli.utilities.utils import get_custom_repo_url
+from utility.log import Log
+
+log = Log(__name__)
```
Comment (vamahaja) [inline tests/cephadm/test_package_validation.py:6]: Unused variable, please remove

Comment (adityaramteke) [reply tests/cephadm/test_package_validation.py]: Removed the variable

**Review by vamahaja — APPROVED** [review]

**Pull request merged**